lib/tests-from-pull-request.js

```diff
@@ -1,32 +1,38 @@
-const fs = require('fs')
-const readline = require('readline')
+module.exports = getTestsFromPR;
 
-module.exports = getTestsFromPR
-
-function getTestsFromPR (context) {
+function getTestsFromPR(context) {
   return Promise.resolve()
-    .then(() => ['Hi Or'])
-  // .then(() => _getTestsFromPR(context))
+    .then(() => _getTestsFromPR(context));
 }
 
-function _getTestsFromPR (context) {
-  const pull = context.issue()
-  const {repos, pullRequests} = context.github
-  // todo: code to get file contect
-  let newTestLines = []
-  let PRfiles
+function _getTestsFromPR(context) {
+  let PRfiles;
+  const pull = context.issue();
+  const {pullRequests} = context.github;
+
   return pullRequests.getFiles(pull)
     .then((res) => res.data)
     .then((files) => PRfiles = files)
-    .then(() => console.log('FILES FROM PR:\n', PRfiles))
+    .then(() => PRfiles.map((file) => file.patch))
+    .then((patches) => patches.join('\n'))
+    .then(_getTestNamesFromPatch);
+}
+
 
-  // let line = ''
-  // if (line.startsWith('+')) {
-  //   line = line.slice(1).trim() // remove the + and spaces
-  //   if (line.trim().startsWith('it(')) {
-  //     newTestLines.push(line)
-  //   }
-  // }
+function _getTestNamesFromPatch(patch) {
+  let newTestLines = [];
+  const parsedPatch = patch.split('\n');
+  parsedPatch.forEach((line) => {
+    if (line.startsWith('+')) {
+      line = line.slice(1).trim(); // remove the + and spaces
+      if (line.trim().startsWith('it(')) {
+        const _testRegex = new RegExp('(?<=\\(\').+?(?=\'\\))', 'g');
+        const _testText = line.match(_testRegex)[0];
+        const resultLine = ['it', _testText].join(' ');
```

Comment:
- `line.match` will return the entire matches but won't return captured groups (which is what you actually want). please use [`regex.exec(str)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec) instead.

- change the string array to a template literal (using backticks)

- please add a line separation between all the prep up to line 32 and the main event (line 32 itself).